internal/service/databaseuser/model_database_user_test.go

```diff
@@ -0,0 +1,347 @@
+		expectedError       bool
+	}{
+		{
+			name:                "Success CloudDatabaseUser",
+			TfDatabaseUserModel: *getDatabaseUserModel(rolesSet, labelsSet, scopesSet),
+			expectedResult:      cloudDatabaseUser,
+			expectedError:       false,
+		},
+		{
+			name:                "Roles fail",
+			TfDatabaseUserModel: *getDatabaseUserModel(wrongRoleSet, labelsSet, scopesSet),
+			expectedError:       true,
+		},
+		{
+			name:                "Labels fail",
+			TfDatabaseUserModel: *getDatabaseUserModel(rolesSet, wrongLabelSet, scopesSet),
+			expectedError:       true,
+		},
+		{
+			name:                "Scopes fail",
+			TfDatabaseUserModel: *getDatabaseUserModel(rolesSet, labelsSet, wrongScopeSet),
+			expectedError:       true,
+		},
+	}
+
+	for i, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			resultModel, err := databaseuser.NewMongoDBDatabaseUser(context.Background(), &testCases[i].TfDatabaseUserModel)
+
+			if (err != nil) != tc.expectedError {
+				t.Errorf("Case %s: Received unexpected error: %v", tc.name, err)
+			}
+			if !reflect.DeepEqual(resultModel, tc.expectedResult) {
+				t.Errorf("created terraform model did not match expected output")
+			}
+		})
+	}
+}
+
+func TestNewTfDatabaseUserModel(t *testing.T) {
+	testCases := []struct {
+		expectedResult  *databaseuser.TfDatabaseUserModel
+		currentModel    databaseuser.TfDatabaseUserModel
+		sdkDatabaseUser *admin.CloudDatabaseUser
+		name            string
+		expectedError   bool
+	}{
+		{
+			name:            "Success TfDatabaseUserModel",
+			sdkDatabaseUser: cloudDatabaseUser,
+			currentModel:    databaseuser.TfDatabaseUserModel{Password: types.StringValue(password)},
+			expectedResult:  getDatabaseUserModel(rolesSet, labelsSet, scopesSet),
+			expectedError:   false,
+		},
+	}
+
+	for i, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			resultModel, err := databaseuser.NewTfDatabaseUserModel(context.Background(), &testCases[i].currentModel, testCases[i].sdkDatabaseUser)
+
+			if (err != nil) != tc.expectedError {
+				t.Errorf("Case %s: Received unexpected error: %v", tc.name, err)
+			}
+			if !reflect.DeepEqual(resultModel, tc.expectedResult) {
+				t.Errorf("created terraform model did not match expected output")
+			}
+		})
+	}
+}
+
+func TestNewMongoDBAtlasScopes(t *testing.T) {
+	testCases := []struct {
+		name           string
+		currentScopes  []*databaseuser.TfScopeModel
+		expectedResult []admin.UserScope
+	}{
+		{
+			name:           "Success TfScopeModel",
+			currentScopes:  []*databaseuser.TfScopeModel{&tfScope},
+			expectedResult: []admin.UserScope{sdkScope},
+		},
+		{
+			name:           "Empty TfScopeModel",
+			currentScopes:  []*databaseuser.TfScopeModel{},
+			expectedResult: []admin.UserScope{},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			resultModel := databaseuser.NewMongoDBAtlasScopes(tc.currentScopes)
+
+			if !reflect.DeepEqual(resultModel, tc.expectedResult) {
```

Comment:
@lantoli  didn't you suggest to use a different comparison way? Not sure if this might be related

Reply:
@marcosuma you are right! Forgot to use the assert from testify library. Changed it.